kalite/topic_tools/__init__.py

```diff
@@ -81,26 +82,32 @@ def recurse_nodes(node):
             else:
                 # By default this is very charitable, assuming if something has not been annotated
                 # it is available - needs to be updated for exercises.
-                if content_cache.get(node.get("id"), {}).get("languages", True):
+                if content_cache.get(node.get("id"), {}).get("available", True):
                     node["available"] = True
 
-        recurse_nodes(TOPICS[channel])
+            # Translate everything for good measure
+            translation.activate(i18n.lcode_to_django_lang(language))
+            node["title"] = _(node.get("title", ""))
+            node["description"] = _(node.get("description", ""))
+            translation.deactivate()
+
+        recurse_nodes(TOPICS[channel][language])
         if settings.DO_NOT_RELOAD_CONTENT_CACHE_AT_STARTUP:
             try:
-                with open(TOPICS_FILEPATHS.get(channel) + ".cache", "w") as f:
-                    json.dump(TOPICS[channel], f)
+                with open(TOPICS_FILEPATHS.get(channel) + "_" + language + ".cache", "w") as f:
+                    json.dump(TOPICS[channel][language], f)
             except IOError as e:
                 logging.warn("Annotated topic cache file failed in saving with error {e}".format(e=e))
 
-    return TOPICS[channel]
+    return TOPICS[channel][language]
 
 
 NODE_CACHE = None
 CACHE_VARS.append("NODE_CACHE")
-def get_node_cache(node_type=None, force=False):
+def get_node_cache(node_type=None, force=False, language="en"):
```

Comment:
what are the negative effects of just setting the default value of the language parameter to `settings.LANGUAGE`?